sql/core/src/main/scala/org/apache/spark/sql/execution/datasources/v2/parquet/ParquetScanBuilder.scala

```diff
@@ -68,25 +73,12 @@ case class ParquetScanBuilder(
         // The rebase mode doesn't matter here because the filters are used to determine
         // whether they is convertible.
         RebaseSpec(LegacyBehaviorPolicy.CORRECTED))
-      parquetFilters.convertibleFilters(pushedDataFilters).toArray
+      parquetFilters.convertibleFilters(dataFilters).toArray
     } else {
       Array.empty[Filter]
```

Comment:
shouldn't we always return the input `dataFilters`?

Reply:
@cloud-fan `FileScanBuilder#pushDataFilters` should return the filters currently supported by the datasource. I refer to the implementation of `OrcScanBuilder` and `JsonScanBuilder`. And this is consistent with the previous parquet implementation logic.

`FileScanBuilder#pushFilters` should always return `dataFilters`.